examples/api/test/widgets/animated_switcher/animated_switcher.0_test.dart

```diff
@@ -0,0 +1,100 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/widgets/animated_switcher/animated_switcher.0.dart'
+    as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('Increments counter on button tap', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const example.AnimatedSwitcherExampleApp(),
+    );
+
+    int counter = 0;
+
+    expect(find.text('$counter'), findsOneWidget);
+
+    while (counter < 10) {
+      // Tap on the button to increment the counter.
+      await tester.tap(
+        find.ancestor(
+          of: find.text('Increment'),
+          matching: find.byType(ElevatedButton),
+        ),
+      );
+      await tester.pumpAndSettle();
+
+      counter += 1;
+
+      expect(find.text('${counter - 1}'), findsNothing);
+      expect(find.text('$counter'), findsOneWidget);
+    }
+  });
+
+  testWidgets('Animates counter change', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const example.AnimatedSwitcherExampleApp(),
+    );
+
+    final Finder zeroTransitionFinder = find.ancestor(
+      of: find.text('0'),
+      matching: find.byType(ScaleTransition),
+    );
+    final Finder oneTransitionFinder = find.ancestor(
+      of: find.text('1'),
+      matching: find.byType(ScaleTransition),
+    );
+
+    expect(zeroTransitionFinder, findsOneWidget);
+    ScaleTransition zeroTransition = tester.widget(zeroTransitionFinder);
+    expect(zeroTransition.scale.value, equals(1.0));
+
+    expect(oneTransitionFinder, findsNothing);
+
+    // Tap on the button to increment the counter.
+    await tester.tap(
+      find.ancestor(
+        of: find.text('Increment'),
+        matching: find.byType(ElevatedButton),
+      ),
+    );
+    await tester.pump();
+
+    expect(zeroTransitionFinder, findsOneWidget);
+    zeroTransition = tester.widget(zeroTransitionFinder);
+    expect(zeroTransition.scale.value, equals(1.0));
+
+    expect(oneTransitionFinder, findsOneWidget);
+    ScaleTransition oneTransition = tester.widget(oneTransitionFinder);
+    expect(oneTransition.scale.value, equals(0.0));
+
+    // Advance animation to the middle.
+    await tester.pump(
+      example.AnimatedSwitcherExampleApp.duration ~/ 2,
```

Comment:
If you want pump one half duration, please simply put 250ms value here in single